sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +684,154 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
   /**
    * Get all active applications of an institution location
-   * with application_status is completed
+   * with application status completed and respective archive status.
    * @param locationId location id .
+   * @param paginationOptions
+   * @param applicationStatus View status of applications requested by user.
    * @returns Student Active Application list.
    */
-  async getActiveApplications(locationId: number): Promise<Application[]> {
+  async getActiveApplications(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+    applicationStatus: ApplicationStatus,
+  ): Promise<PaginatedResults<Application>> {
     // TODO: there are two similar methods to get one and many records for the same list/details getActiveApplication and getActiveApplications. Can we use only one?
-    return this.repo
+    const activeApplicationQuery = this.repo
       .createQueryBuilder("application")
       .select([
         "application.applicationNumber",
         "application.id",
         "currentAssessment.id",
-        "application.applicationStatus",
         "offering.studyStartDate",
         "offering.studyEndDate",
         "student.id",
         "user.firstName",
         "user.lastName",
+        "application.isArchived",
+        "studentScholasticStanding.id",
       ])
       .leftJoin("application.currentAssessment", "currentAssessment")
       .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "currentAssessment.studentScholasticStanding",
+        "studentScholasticStanding",
+      )
       .innerJoin("application.student", "student")
       .innerJoin("student.user", "user")
       .where("application.location.id = :locationId", { locationId })
       .andWhere("application.applicationStatus = :applicationStatus", {
         applicationStatus: ApplicationStatus.completed,
-      })
-      .orderBy("application.applicationNumber", "DESC")
-      .getMany();
+      });
+
+    // Available applications are records that have no 'scholastic standing reported' and are not 'archived'.
+    if (applicationStatus === ApplicationStatus.available) {
+      activeApplicationQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("studentScholasticStanding.id IS NULL").andWhere(
+            "application.isArchived = :isArchived",
+            {
+              isArchived: false,
+            },
+          );
+        }),
+      );
+    }
+
+    // Unavailable applications are records that are either 'completed (scholastic standing reported but not yet archived)' or 'archived'.
+    if (applicationStatus === ApplicationStatus.unavailable) {
```

Comment:
This condition should be probably simplified to `!archived`. This method in general is concerned with retrieving archived and not archived applications. Completed/available are sub-statuses from the archived ones.